src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,100 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exeptions.InvalidCredentialsException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final int LOGIN_AND_PASSWORD_LEN = 6;
+    private static final int AGE = 18;
+    private StorageDaoImpl storageDao = new StorageDaoImpl();
+    private RegistrationServiceImpl registrationService;
+    private User defaultUser;
+
+    @BeforeEach
+    public void createDefaultUser() {
+        registrationService = new RegistrationServiceImpl();
+        defaultUser = new User(1L, "default", "password", 18);
+    }
+
+    @Test
+    public void register_userWithCorrectData_Ok() {
+        User expected;
+        try {
+            expected = registrationService.register(defaultUser);
+        } catch (InvalidCredentialsException e) {
+            throw new RuntimeException("User is registered");
+        }
+        assertEquals(defaultUser, expected);
+    }
+
+    @Test
+    public void register_sameUsers_notOk() {
+        storageDao.add(defaultUser);
+        assertThrows(InvalidCredentialsException.class,
+                () -> registrationService.register(defaultUser));
+    }
+
+    @Test
+    public void register_loginLessThan6_notOk() {
+        StringBuilder login = new StringBuilder();
+        for (int i = 1; i < LOGIN_AND_PASSWORD_LEN; i++) {
+            login.append("a");
```

Comment:
Don’t complicate your test case with additional logic. Create a variable with an invalid login (the same for other test cases)